File: src/main/java/com/spring/spring_init/oauth/handler/OAuth2AuthenticationFailureHandler.java
package com.spring.spring_init.oauth.handler;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.IOException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class OAuth2AuthenticationFailureHandler extends SimpleUrlAuthenticationFailureHandler {

    @Value("${common.full-domain}")
    private String domain;

    @Override
    public void onAuthenticationFailure(
        HttpServletRequest request,
        HttpServletResponse response,
        AuthenticationException exception
    ) throws IOException {

//        response.sendRedirect("/oauth/authorize/kakao");
        //프로젝트 구현 방식에 따라 알맞게 구현
        getRedirectStrategy().sendRedirect(request, response, domain+"/fail");
    }
}

File: src/main/java/com/spring/spring_init/oauth/handler/OAuth2AuthenticationFailureHandler.kt
package com.spring.spring_init.oauth.handler

import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler
import org.springframework.stereotype.Component
import java.io.IOException

@Component
class OAuth2AuthenticationFailureHandler : SimpleUrlAuthenticationFailureHandler() {

    companion object {
        private val log = LoggerFactory.getLogger(OAuth2AuthenticationFailureHandler::class.java)
    }

    @Value("\${common.full-domain}")
    private lateinit var domain: String

    @Throws(IOException::class)
    override fun onAuthenticationFailure(
        request: HttpServletRequest,
        response: HttpServletResponse,
        exception: AuthenticationException
    ) {
//        response.sendRedirect("/oauth/authorize/kakao")
        //프로젝트 구현 방식에 따라 알맞게 구현
        redirectStrategy.sendRedirect(request, response, "$domain/fail")
    }
}
